### DISCUSSION TITLE
What is the plan?

### DISCUSSION THREAD

**Discussion by moay** [CodeDredd/pinia-orm#4] [General]
Hi everybody!

First of all: Thanks for tackling the task of building VuexORM for Pinia. 👏 

What is the plan with this project? Development looks a bit stalled, but are there plans to really make this project something viable? It looks like there are bigger efforts needed to make this production ready?

I'm wondering how people work with frontend representations of relational dbs without native ORM in the big store projects. Maybe I'm missing something here and there is a good alternative to building an ORM on top of Pinia?

**Comment by CodeDredd** [comment]
@moay Hey,

thanks for the interest. Right now i am on vaccation. But will soon continue to work a bit on open source again.

My Road map which i have in mind.
Well i am already working on finishing tests and bit more code clean up so i have a version 1 which should be only the basic like vuex-orm. The next thing is also to migrate some plugins which were already asked like "vuex-orm-axios". 

After that i want to move and make v2. If more people are interested in this plugin i am going to start RFCs.

What i have in mind for v2 is:

- better devtools integration
- next level data fetching & caching (Going to explain it later ^^)

**Reply by tintin10q** [reply]
Please also add belongsToMany

**Reply by CodeDredd** [reply]
@tintin10q Your please is coming soon ;-) #79

**Reply by vesper8** [reply]
@CodeDredd First off thanks a ton for porting vuex-orm to pinia!! I've been using vuex-orm for many years and now that I have a new project that requires me to use Pinia, I was so happy to find your project!

I've also been using `@vuex-orm/plugin-axios` and `@vuex-orm/plugin-search` and I wanted to know where you were at as far as porting the functionality provided by `@vuex-orm/plugin-axios`? I see it mentioned in a few of these discussions but don't see anything in the Documentation or main Readme so am I right to assume this functionality is not yet available? Any chance you have any kind of ETA on when it may be added? Having a permanent bridge between the database backend and the client-side ORM is quite important to my new project.

Many thanks!!

**Reply by CodeDredd** [reply]
@vesper8 Thanks. Well i have some todos on the list for pinia-orm. At the moment i was busy with work, kids and being happy not to be ill. ^^
My thoughts on what i work.
- Making examples for axios, vue-query, i18n, etc. and how to use them now (WIIP - #999)
- Thinking on the next iteration of pinia-orm v2. I think i try to decouple pinia as package from the orm so its better usable with dexie or usable with vuex.

I am really happy about every feedback or code contribution. Also every bit of support like spending me a cup of ☕ helps ❤️ .

**Reply by vesper8** [reply]
@CodeDredd Thanks for the update and sorry for the late reply. Happy to see there has been some active development and enhancements added since February.

At this point I'm still on hold to migrate from Vuex to Pinia because of my heavy use of vuex-orm and my heavy use of the functionalities provided by both https://github.com/vuex-orm/plugin-axios and https://github.com/vuex-orm/plugin-search.

Can you comment on any plans to add this kind of functionality, to the core or as plugins, to pinia-orm in the near future?

Many thanks!

**Reply by vesper8** [reply]
@CodeDredd It's so awesome that you've now released the axios plugin! Do you have any comment on vuex-orm's search plugin and if there is currently a way to achieve the same search functionality (without using algolia, fully locally) ? I also use the search plugin a lot and it would be so great if it was also added to pinia-orm, I think this may be the last missing piece to allow everyone to migrate fully from vuex-orm.

Many thanks!

**Comment by tintin10q** [comment]
Ple

**Reply by CodeDredd** [reply]
Ok putting it on the wishlist ;-) ....i know it also is missing already in vuex-orm-next. There was i think also a prototype how to implement it. Maybe i just take that into this version since version 1 should just be a compatible version to vuex-orm models.

**Comment by paolodina** [comment]
@CodeDredd 
Well, I just say: thanks for your efforts on this project and keep it going too! I have a new project and will use pinia-orm soon, I need this kind of tool.

> next level data fetching & caching (Going to explain it later ^^)

At this regard, when I didn't even know about this discussion, I was considering to complement pinia-orm with vue-query. Do you think (or, would you consider) it could be convenient an integration instead of writing a such thing from scratch? It seems to be [a complex task](https://vue-query.vercel.app/#/?id=motivation).

An ORM for Pinia with a proper data fetching layer as an all-in-one solution would be a dream.

**Reply by CodeDredd** [reply]
@paolodina First thanks! 
Second your idea with vue-query sounds great. From the first look it seems that you can use pinia-orm and vue-query together without any changes. But maybe i am wrong. But i should defently start a documentation part with examples for it.
Yeah the caching thing i already thought some hours about it and because of SSR its really not simple and vue-query comes here really handy.

**Reply by CodeDredd** [reply]
@paolodina And thanks for the cup of ☕ . Every bit helps ❤️ 

It seems i really going to need parts in the docs for vue-query, axios and graphql. The good thing is that i am almost finished with the orm part.
I will also include the same caching mechanism in pinia-orm like in vue-query, reat-query, vuex-orm-cache, ... 
At the end vue-query, or whatever api solution is taken should be wokring together with pinia-orm.

**Reply by CodeDredd** [reply]
@paolodina One question 😃 
Why are you using `vue-query` and not [tanstack-query](https://github.com/TanStack/query) ? I haven't yet used any of them but it seems like there are all the same. Maybe you know the advantage of `vue-query` because if i look at the size then defently tanstack wins by far.

**Reply by CodeDredd** [reply]
Caching incoming 🥳 https://github.com/CodeDredd/pinia-orm/pull/272

**Comment by paolodina** [comment]
Good job on the new cache!

> Why are you using vue-query and not [tanstack-query](https://github.com/TanStack/query)

I still haven't tried any of those, but this was enough to determine my choice:

![image](https://user-images.githubusercontent.com/1157401/187280449-8447795e-d4aa-4c04-b5a4-d306b9603b88.png)

Solid, Svelte and Vue adapter: coming soon.

For the bundle size I don't see it as a problem ([13kb](https://bundlephobia.com/package/react-query@3.39.2) VS [16.7kb](https://bundlephobia.com/package/vue-query@1.26.0)).

> It seems i really going to need parts in the docs for vue-query, ...

Documentation plays an important part and would be really great (but not strictly essential) to have *that* missing part :smile:

**Comment by jhercog** [comment]
@CodeDredd 
First, I'd like to join others in thanking you for all the efforts on making and maintaining this project. :)
Secondly, I'm also interested in vue-query and/or axios integration. Are there any news on expanding the docs with detailed explanation on how to do it manually or (in the best case) making the extension plugins?

Keep up the good work.

**Reply by CodeDredd** [reply]
@jhercog Many Thanks for the support ❤️ .
Well i already thought about the concept of how to make it. It should be a composable, which then can be used to extend the repository.
I described here for graphql for example: https://github.com/CodeDredd/pinia-orm/discussions/1013
